### PR TITLE
feat!: move diagnostics_module from localization_util to unverse_utils

### PR DESCRIPTION
**Pull request by kminoda**
## Description

`DiagnosticsModule` in `localization_util` is useful. Thus I would like to move it from the package to `universe_utils`

## Related links

None

## How was this PR tested?

Added unit test for the module

## Notes for reviewers

None.

## Interface changes

Package include path would change.

## Effects on system behavior

None.

### CODE REVIEW

File: common/autoware_universe_utils/include/autoware/universe_utils/ros/diagnostics_module.hpp
Comment (soblin) [inline common/autoware_universe_utils/include/autoware/universe_utils/ros/diagnostics_module.hpp:60]: Please remove these declarations because they do not make sense unless you declare them as "extern template" and define them in cpp files as specializations.

File: common/autoware_universe_utils/include/autoware/universe_utils/ros/diagnostics_module.hpp
Comment (soblin) [inline common/autoware_universe_utils/include/autoware/universe_utils/ros/diagnostics_module.hpp:27]: IMO DiagnosticInterface is better

File: common/autoware_universe_utils/src/ros/diagnostics_module.cpp
Comment (SakodaShintaro) [inline common/autoware_universe_utils/src/ros/diagnostics_module.cpp:79]: @YamatoAndo 
Is this change consistent with your original intent?

Comment (YamatoAndo) [reply common/autoware_universe_utils/src/ros/diagnostics_module.cpp:79]: no

Comment (YamatoAndo) [reply common/autoware_universe_utils/src/ros/diagnostics_module.cpp:79]: If you want to do so, this if statement will always be satisfied and is unnecessary, so please remove it.
Additionally, ensure that "; " is not appended when the message is empty.

Also, it would be better to revise the code at line 102.
like this,
```
  if (diagnostics_msg.status.at(0).message.empty()) {
    if (diagnostics_msg.status.at(0).level == diagnostic_msgs::msg::DiagnosticStatus::OK) {
    diagnostics_msg.status.at(0).message = "OK";
    }
    else if (diagnostics_msg.status.at(0).level == diagnostic_msgs::msg::DiagnosticStatus::WARN) {
    diagnostics_msg.status.at(0).message = "WARN";
    }
    else if (diagnostics_msg.status.at(0).level == diagnostic_msgs::msg::DiagnosticStatus::ERROR) {
    diagnostics_msg.status.at(0).message = "ERROR";
    }
    else if (diagnostics_msg.status.at(0).level == diagnostic_msgs::msg::DiagnosticStatus::STALE) {
    diagnostics_msg.status.at(0).message = "STALE";
    }
  }
```

Comment (kminoda) [reply common/autoware_universe_utils/src/ros/diagnostics_module.cpp]: Sorry that was a complete mistake that I introduced while I played around with the code. Thanks for the catch. Fixed: https://github.com/autowarefoundation/autoware.universe/pull/9714/commits/113d770fb442309642d869971a78472eda2d005c

File: common/autoware_universe_utils/include/autoware/universe_utils/ros/diagnostics_module.hpp
Comment (kminoda) [reply common/autoware_universe_utils/include/autoware/universe_utils/ros/diagnostics_module.hpp]: Fixed: https://github.com/autowarefoundation/autoware.universe/pull/9714/commits/b7a90d82971fa10768ed0d5525df6b7bafb9bccb

File: common/autoware_universe_utils/include/autoware/universe_utils/ros/diagnostics_module.hpp
Comment (kminoda) [reply common/autoware_universe_utils/include/autoware/universe_utils/ros/diagnostics_module.hpp]: How about this? https://github.com/autowarefoundation/autoware.universe/pull/9714/commits/c981570b25bb48a1bd426b6e162cb531db3cc640

**Comment by kminoda** [comment]
Thank you to all the reviewers. I've addressed all the comments.

**Review by SakodaShintaro — APPROVED** [review]
I checked the following test items:

- [x] `planning_simulator` with [sample_map](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/planning-simulation/)
- [x] `logging_simulator` with [sample_rosbag](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/rosbag-replay-simulation/)
- [x] `logging_simulator` with [awsim_gt_data(TIER IV INTERNAL)](https://drive.google.com/file/d/1BtH3Ry5lu-h85GHM-sYq_jNJLw-s0re6/view?usp=drive_link)
- [x] `e2e_simulator` with [AWSIM v1.3.1](https://github.com/tier4/AWSIM/releases/tag/v1.3.1)

I also verified that the `/diagnostics` topic had the appropriate information.

LGTM, but please get soblin-san's approval before merging.

**Comment by SakodaShintaro** [comment]
Sorry, [pre-commit.ci](https://results.pre-commit.ci/run/github/429651616/1735086786.OxvDZpz3Tk6qSwKa7r6cdA) is failing.

**Comment by kminoda** [comment]
@SakodaShintaro Thanks, fixed the precommit.

**Review by soblin — APPROVED** [review]

**Pull request merged**
